### ISSUE TITLE
[FR] Better folder note count

### ISSUE THREAD

**Issue by SPiCaRiA** [isaaclyman/novel-word-count-obsidian#35]
Hi, I came from your comment under this [issue](https://github.com/ozntel/file-explorer-note-count/issues/13#issuecomment-1294151209) of the File Explorer Note Count plug-in, and I'm looking forward to this being a better substitute of it!

However, one major issue that keeps us from that goal is the folder note exclusion. Also, instead of introducing a new dependency (File Explorer Note Count relies on the Folder Note Core plug-in), we can make it simple and only treat notes with the same name as the folder as folder notes. What do you think about this?

Thanks!

**Comment by isaaclyman** [comment]
Hi, thanks for the suggestion. Right now I'm leaning against it for a few reasons:

- Folder Note Core is not a core plugin, and I try to avoid going too far out of my way for compatibility with community plugins.
- Having a note with the same name as the folder is a pretty common thing, and people will have different expectations around it. I do this a lot, personally, and it would be a problem if Novel Word Count didn't count the note.
- Even people who use Folder Note Core may expect the folder note to be counted. After all, it exists, it has text content, and Obsidian can read it. It makes more sense to count it IMO.

**Comment by isaaclyman** [comment]
My automatic response to "let's just make a setting" is always no. It's a word count plugin, not an airplane cockpit. If I were to add every setting someone asked for, it would be so complicated that new users would just give up.

If a lot of people ask for this I will revisit.

Perhaps an alternative way to do this without complicating the settings is with a file naming convention. (e.g. "all files with the extension .xcount.md will not be counted.") With Folder Note Core, does it pick the filename for you or can you choose it yourself?

**Comment by SPiCaRiA** [comment]
> My automatic response to "let's just make a setting" is always no. It's a word count plugin, not an airplane cockpit. If I were to add every setting someone asked for, it would be so complicated that new users would just give up.
> 
> If a lot of people ask for this I will revisit.
> 
> Perhaps an alternative way to do this without complicating the settings is with a file naming convention. (e.g. "all files with the extension .xcount.md will not be counted.") With Folder Note Core, does it pick the filename for you or can you choose it yourself?

TL;DR: we can start with "it picks the filename (same as the folder's name) for you" before we support more strategies.

I see, that makes a lot of sense. For folder note plug-ins in general (there are a bunch of plug-ins for this purpose so I would suggest not sticking onto a specific one), there are 3 strategies AFAIK:

1. Same name as the folder, inside the folder:
```
- Folder
  - Folder.md
```
2. Same name as the folder, outside the folder (i.e., both under the parent folder):
```
- ParentFolder
  - Folder
  - Folder.md
```
3. Specific and fixed name, inside the folder (let's say the name of all folder notes is `__foldernote__`)
```
- Folder
  - __foldernote__.md
```
(4. Specific name outside the folder does not work because of naming conflicts, if the parent folder has more than one subfolder that all have folder notes).

Strategy 1 is the most popular one, so I said we can start with this one.

**Comment by isaaclyman** [comment]
Okay. So the plugin picks the filename for you and you're not allowed to change it, which means the only way to support exclusion of folder notes is by convention + an opt-in setting.

For that reason I'm declining this feature for now. If more requests come in over time I'll give it another look.

**Comment by greetclammy** [comment]
I'd like to also request the option to exclude folder notes from the folder note count 👍
